config_helper.py

```diff
@@ -0,0 +1,300 @@
+    like fonts and images are not preloaded into the resources cache. Rather they
+    are processed and loaded as needed.
+From within a CardGen script, the specific function should be called to load
+    a particular type from the resources cache. E.g. To load a font, the getfont()
+    function should be called.
+config_helper supports a hierarchy of default values in both the config file and
+    resources cache. Every function to be called from the CardGen script should
+    include both the card name and the card type. This way, config_helper will
+    check the resources cache or config file in the following order of sections
+    for the given option: Card name -> Card type -> Card Defaults
+If the resource is not found in the cache (for some options), config_helper
+    will attempt to load that resource from the config and add it to the cache.
+'''
+
+import os
+import PIL_Helper as ph
+from ConfigParser import SafeConfigParser, NoOptionError
+
+config = None
+resources_cache = None
+DEFAULT_SECTION = "Card Defaults"
+
+pjoin = os.path.join
+
+def LoadConfig(folder, card_set, filename):
+    '''
+    @param str folder: The game folder, e.g. TSSSF
+    @param str card_set: The folder within the game folder that stores the
+        given card set. e.g., Core Deck 1.1.5
+    @param str filename: The name of the config files to be loaded. Both config
+        files must have the same name.
+
+    Loads a required config file from the game folder, passing it back to the main
+        function. Can also load an optional config file in the card_set folder, and
+        that config can overwrite anything in the first config. This is useful for
+        making small tweaks to the config file that apply only to that card set.
+
+    @return None: The config file and resources cache will be saved in this module.
+        No other module should be handling these objects.
+    '''
+    global config, resources_cache
+    # Init config object
+    config = SafeConfigParser()
+    # Load main config
+    config.read(pjoin(folder, filename))
+    # Load possible alternate config
+    alt_config_path = pjoin(folder, card_set, filename)
+    if os.path.exists(alt_config_path):
+        config.read(alt_config_path)
+    # Initialize the resources_cache
+    resources_cache = {}
+
+def get(option, section="Settings", val_type="str", default=None):
+    '''
+    @param str option: The option to pull the value for
+    @param str section: The config section to pull from
+    @param str val_type: The value type to try to cast the config option to.
+        Accepted options: str (default), int, float, bool
+
+    Convenience function for allowing 
+    Attempts to return the value for the given option, first from a section
+        matching card_name, then from a section matching card_type, then
+        from the Defaults section. If none of the sections (if they exist)
+        have that option, then None is returned.
```

Comment:
This documentation is out of date and needs to be updated. get() only looks at a particular section, right now. Other functions will use find_files_in_config before calling get(), so they also know where in the resources cache to add the new data.